src/overlays/actors/ovl_Dm_Char08/z_dm_char08.c

```diff
@@ -28,71 +53,1002 @@ const ActorInit Dm_Char08_InitVars = {
     (ActorFunc)DmChar08_Draw,
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80AB1764[] = {
+#include "overlays/ovl_Dm_Char08/ovl_Dm_Char08.c"
+
+AnimationInfo sAnimationInfo[7] = {
+    { &object_kamejima_Anim_0048B0, 1.0f, 0.0f, -1.0f, 0, -24.0f },
+    { &object_kamejima_Anim_006980, 1.0f, 0.0f, -1.0f, 0, -24.0f },
+    { &object_kamejima_Anim_012260, 1.0f, 0.0f, -1.0f, 0, -24.0f },
+    { &object_kamejima_Anim_0100CC, 1.0f, 0.0f, -1.0f, 0, -24.0f },
+    { &object_kamejima_Anim_0047B8, 1.0f, 0.0f, -1.0f, 0, -24.0f },
+    { &object_kamejima_Anim_0119D4, 1.0f, 0.0f, -1.0f, 0, -24.0f },
+    { &object_kamejima_Anim_014E8C, 1.0f, 0.0f, -1.0f, 0, -24.0f },
+};
+
+static InitChainEntry sInitChain[] = {
     ICHAIN_F32(uncullZoneForward, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneScale, 4000, ICHAIN_CONTINUE),
     ICHAIN_F32(uncullZoneDownward, 4000, ICHAIN_STOP),
 };
 
-#endif
+void DmChar08_UpdateEyes(DmChar08* this) {
+    switch (this->eyeMode) {
+        case EYEMODE_0:
+            this->eyeIndex = 0;
+            if (this->blinkTimer > 0) {
+                this->blinkTimer--;
+            } else {
+                this->blinkTimer = 0;
+            }
+            if (this->blinkTimer < 4) {
+                this->eyeIndex = this->blinkTimer;
+            }
+            if (this->blinkTimer == 0) {
+                this->blinkTimer = Rand_S16Offset(30, 30);
+            }
+            break;
+        default:
+            break;
+        case EYEMODE_1:
+            this->eyeIndex = 4;
+            if (this->blinkTimer > 0) {
+                this->blinkTimer--;
+            } else {
+                this->blinkTimer = 0;
+            }
+            if (this->blinkTimer < 4) {
+                this->eyeIndex = this->blinkTimer + 4;
+            }
+            if (this->blinkTimer == 0) {
+                this->blinkTimer = Rand_S16Offset(30, 30);
+            }
+            break;
+        case EYEMODE_2:
+            this->eyeIndex = 2;
+            break;
+        case EYEMODE_3:
+            this->eyeIndex = 4;
+            break;
+        case EYEMODE_5:
+            this->eyeIndex = 8;
+            break;
+    }
+}
+
+void DmChar08_SetAnimation(SkelAnime* skelAnime, AnimationInfo* entry, u16 arg2) {
```

Comment:
```suggestion
void DmChar08_SetAnimation(SkelAnime* skelAnime, AnimationInfo* entry, u16 index) {
```
Looks like it indexes in to the entry array of arg1.